src/QMCWaveFunctions/BsplineFactory/contraction_helper.hpp

```diff
@@ -0,0 +1,72 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 QMCPACK developers.
+//
+// File developed by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//
+// File created by: Ye Luo, yeluo@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
```

Comment:
One of these functions is only used in test code and two only in SplineR2RAdoptor but they are just floating around here with no scope. Where are they going?

Reply:
They are all used in the C2R, C2C and R2R adopters and it is the reason of the move.

Reply:
I actually didn't pay attention to move the test. Can I do it in a later PR?

Reply:
ok why aren't they scoped into those? Then they should go into the base class.  
call via
```C++
BaseType::v_m_v(...)
```

Reply:
Did you mean SplineAdoptorBase? I can move them in.

Reply:
Actually not. The Base class was not specialized to do 3D by definition.

Reply:
Yes that's what I meant. The unit test can stay put if it can adapt.

Reply:
Or write an issue for it. And disable it for the moment.

Reply:
An issue is created.